projboard/tests/test_like.py

```diff
@@ -1,50 +1,58 @@
 import pytest
 from projboard.models import Like, User, Article, Subject
 
-U = ['usertest@gmail.co.il', '123456', 'John Doe', 'USER_TEST']
-USER_TEST = User(email=U[0], password=U[1], name=U[2], nickname=U[3])
+USER_DETAILS = ['usertest@gmail.co.il', '123456', 'John Doe', 'USER_TEST']
+USER_TEST = User(email=USER_DETAILS[0], password=USER_DETAILS[1], name=USER_DETAILS[2], nickname=USER_DETAILS[3])
 
 SUBJECT_TEST = Subject(name='Test')
-A = [USER_TEST, 'World cup', SUBJECT_TEST, 'Messi is better then CR7']
-ARTICLE_TEST = Article(user_id=A[0], title=A[1], subject_id=A[2], content=A[3])
+ARTICLE_DETAILS = [USER_TEST, 'World cup', SUBJECT_TEST, 'Messi is better then CR7']
+ARTICLE_TEST = Article(user_id=ARTICLE_DETAILS[0], title=ARTICLE_DETAILS[1],
+                       subject_id=ARTICLE_DETAILS[2], content=ARTICLE_DETAILS[3])
 
 
 @pytest.fixture
-def generate_like():
-    SUBJECT_TEST.save()
+def generate_like(create_user, create_subject, create_article):
+    Like(user_id=create_user, article_id=create_article).save()
+    return create_user, create_article
+
+
+@pytest.fixture
+def create_user():
     USER_TEST.save()
+    return USER_TEST
+
+
+@pytest.fixture
+def create_subject():
+    SUBJECT_TEST.save()
+    return SUBJECT_TEST
+
+
+@pytest.fixture
+def create_article():
     ARTICLE_TEST.save()
-    Like(user_id=USER_TEST, article_id=ARTICLE_TEST).save()
-    return USER_TEST, ARTICLE_TEST
+    return ARTICLE_TEST
```

Comment:
I'm not sure it's possible to use `.save()` without a db linked, using `@pytest.mark.django_db` or db passed in the argument `create_article(db)` should solve it (if so, the `create_article(db)` should be the right use since you would use the returning value of the function).